### PR TITLE
Add customizable corner radius to accelerated checkout buttons

### PR DESCRIPTION
**Pull request by josemiguel-alvarez**
<details><summary>Screenshot</summary>
image
</details> 

This pull request introduces a new feature to customize the corner radius of the accelerated checkout buttons.. The changes include updates to the `AcceleratedCheckoutButtons` component, individual wallet buttons (`ApplePayButton` and `ShopPayButton`), and related styling and tests. Below is a summary of the most important changes:

### New Feature: Corner Radius Customization

* **`AcceleratedCheckoutButtons` updates**:
  - Added a `cornerRadius` property to allow customization of button corner radii.
  - Introduced a new `.withCornerRadius(_:)` modifier for setting the corner radius of all buttons in the component.

* **Wallet button updates**:
  - Updated `ApplePayButton` and `ShopPayButton` to accept and apply the `cornerRadius` property. This ensures the corner radius is passed down to their internal implementations. [[1]](diffhunk://#diff-42b3097511391a6b01041231a6c2515e9e1ec7adc9bbba24f8d2925758057d51R52-R62) [[2]](diffhunk://#diff-c99c5cb46bc0367a7fdb869e6c349e3cd6ac63cb597a64aeee2ebb36cbf953d5R34-R43)
  - Modified internal wallet button implementations to apply the `cornerRadius` using the `walletButtonStyle` modifier. [[1]](diffhunk://#diff-42b3097511391a6b01041231a6c2515e9e1ec7adc9bbba24f8d2925758057d51L133-R144) [[2]](diffhunk://#diff-c99c5cb46bc0367a7fdb869e6c349e3cd6ac63cb597a64aeee2ebb36cbf953d5L93-R100)

### Styling Enhancements

* **`walletButtonStyle` modifier**:
  - Enhanced to accept an optional `cornerRadius` parameter with a default value of 8. Negative values default to 8.

### Test Coverage

* **Integration tests**:
  - Updated `ApplePayIntegrationTests` to include the `cornerRadius` parameter in button initialization. [[1]](diffhunk://#diff-657e1ad09bee4444091055fdd5b7532c578cdeecaac269355f5b88fdcdb13479L146-R146) [[2]](diffhunk://#diff-657e1ad09bee4444091055fdd5b7532c578cdeecaac269355f5b88fdcdb13479L168-R169)

* **Unit tests**:
  - Added new tests in `ApplePayViewModifierTests` to validate the behavior of the `.withCornerRadius(_:)` modifier, including edge cases for zero and negative values.

### CODE REVIEW

File: Samples/ShopifyAcceleratedCheckoutsApp/ShopifyAcceleratedCheckoutsApp/Views/Components/ButtonSet.swift [hunk 1]
```diff
@@ -62,11 +62,12 @@ struct ButtonSet: View {
                let productVariant = merchandise.asProductVariant
             {
                 CheckoutSection(title: "AcceleratedCheckoutButtons(variantID: quantity:)") {
-                    // Variant-based checkout with separate handlers
+                    // Variant-based checkout with separate handlers and custom corner radius
                     AcceleratedCheckoutButtons(
                         variantID: productVariant.id,
                         quantity: firstVariantQuantity
                     )
+                    .withCornerRadius(24)
```
Comment (markmur) [inline Samples/ShopifyAcceleratedCheckoutsApp/ShopifyAcceleratedCheckoutsApp/Views/Components/ButtonSet.swift:70]: Heads up, the "with" prefixes [diverge from what we use currently for the `CheckoutSheet` usage](https://github.com/Shopify/checkout-sheet-kit-swift?tab=readme-ov-file#swiftui-usage) in Swift UI. I'd recommend `.cornerRadius(24)` instead.

As an aside; it appears `cornerRadius` is actually [deprecated in Swift now](https://developer.apple.com/documentation/swiftui/view/cornerradius(_:antialiased:)) and `.clipShape` is recommended instead 😕

**Review by markmur — COMMENTED** [review]
PR looks good otherwise!

File: Samples/ShopifyAcceleratedCheckoutsApp/ShopifyAcceleratedCheckoutsApp/Views/Components/ButtonSet.swift (hunk 1 above)
Comment (josemiguel-alvarez) [reply Samples/ShopifyAcceleratedCheckoutsApp/ShopifyAcceleratedCheckoutsApp/Views/Components/ButtonSet.swift:70]: We are already using clipShape in this PR, not borderRadius.

`.clipShape(RoundedRectangle(cornerRadius: radius >= 0 ? radius : defaultCornerRadius))`

I'll update the naming.

Comment (josemiguel-alvarez) [reply Samples/ShopifyAcceleratedCheckoutsApp/ShopifyAcceleratedCheckoutsApp/Views/Components/ButtonSet.swift]: Renamed in https://github.com/Shopify/checkout-sheet-kit-swift/pull/331/commits/6db36fbd418fcd431fa776a33ea3c845fe4438e5

**Review by markmur — APPROVED** [review]
